### ISSUE TITLE
Catalog Search Sort by Relevance not working

### ISSUE THREAD

**Issue by muhammadshakeel** [magento/magento2#11005]
### Preconditions

1. Magento Open Source (CE) 2.1.9 with sample data
2. PHP 7.0.22
3. MySQL 5.7.19

### Steps to reproduce

1. Goto Catalog->Catalog->Catalog Search and make sure your Search Engine is selected as MySQL
2. Browse to Store front and search for any term like 'yoga' and see the result sorting. (It is not sorted by relevance).
3. Now update https://github.com/magento/magento2/blob/2.1/app/code/Magento/Store/Model/Store.php#L755 to return integer type like following
```php
public function getId()
{
    return (int) $this->_getData(self::STORE_ID);
}
```
4. Now see the search results, these will be sorted by relevance.

### Expected result

1. Search results should be sorted by relevance. We need to investigate a real cause and fix it.

### Actual result

1. Sort by relevance is not working.

**Comment by magento-engcom-team** [comment]
@muhammadshakeel, thank you for your report.
The issue is already fixed in 2.2.0

**Comment by muhammadshakeel** [comment]
Following can be used as a temporary solution for 2.1.x (hopefully we will get a patch for 2.1.x):

Add following plugin in your di.xml

```xml
<type name="Magento\Store\Model\Store">
    <plugin name="store_plugin_search_hotfix" type="Vendor\Extension\Plugin\StorePlugin" sortOrder="1" />
</type>
```
Add StorePlugin class as follows:

```php
<?php

namespace Vendor\Extension\Plugin;

/**
 * This plugin is used to fix Magento internal issue with search results sorting
 * After typecasting to (int), sort order for 'by relevance' is fixed.
 *
 */
class StorePlugin
{
    public function afterGetId(\Magento\Store\Model\Store $subject, $result)
    {
        return (int) $result;
    }
}
```
Execute `php bin/magento setup:di:compile` command and observe search results.
